plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTest.java

```diff
@@ -127,6 +129,109 @@ public void cancel_success() {
     assertThat(resultDict).isEqualTo(testDict);
   }
 
+  @Test
+  public void assetPackStateUpdatedTest() {
+    doAnswer(
+            invocation -> {
+              AssetPackStateUpdateListener listener =
+                  (AssetPackStateUpdateListener) invocation.getArguments()[0];
+              List<AssetPackState> packStateSequence =
+                  PlayAssetDeliveryTestHelper.createAssetPackStateList();
+              for (AssetPackState packState : packStateSequence) {
+                listener.onStateUpdate(packState);
+              }
+              return null;
+            })
+        .when(assetPackManagerMock)
+        .registerListener(any(AssetPackStateUpdateListener.class));
+
+    ArgumentCaptor<String> signalNameCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<Object> signalArgsCaptor = ArgumentCaptor.forClass(Object.class);
+
+    PlayAssetDelivery testSubject = spy(new PlayAssetDelivery(godotMock, assetPackManagerMock));
+
+    testSubject.registerAssetPackStateUpdatedListener();
+
+    verify(testSubject, times(3))
+        .emitSignalWrapper(signalNameCaptor.capture(), signalArgsCaptor.capture());
+
+    assertThat(signalNameCaptor.getAllValues())
+        .isEqualTo(
+            Arrays.asList(
+                "assetPackStateUpdated", "assetPackStateUpdated", "assetPackStateUpdated"));
```

Comment:
since this string is used three times, I'd make a local variable for it.

Reply:
Modified.